File: parser-ts/parser.ts

import * as fs from 'fs';
import * as path from 'path';

interface LogData {
    [key: string]: {
        pub?: string;
        sub?: string;
    };
}

function parseLogFile(filePath: string): string | null {
    const content = fs.readFileSync(filePath, 'utf-8');
    const match = content.match(/(\d{1,3}(?:,\d{3})*|\d+)\s+msgs\/sec/);
    return match ? match[1].replace(/,/g, '') : null;
}

function getSortKey(dirName: string): number {
    dirName = dirName.toLowerCase();
    if (dirName.endsWith('ms')) {
        return parseInt(dirName.slice(0, -2), 10);
    } else if (dirName.endsWith('s')) {
        return parseInt(dirName.slice(0, -1), 10) * 1000;
    } else if (dirName.endsWith('m')) {
        return parseInt(dirName.slice(0, -1), 10) * 60 * 1000;
    } else {
        return Infinity; // Should not happen
    }
}

function main() {
    const data: LogData = {};
    const currentDir = './';

    const dirs = fs.readdirSync(currentDir).filter(f => fs.statSync(path.join(currentDir, f)).isDirectory());

    for (const dir of dirs) {
        const pubLogPath = path.join(currentDir, dir, 'bench-pub.log');
        const subLogPath = path.join(currentDir, dir, 'bench-sub.log');

        if (fs.existsSync(pubLogPath)) {
            const pubMsgsSec = parseLogFile(pubLogPath);
            if (pubMsgsSec) {
                if (!data[dir]) data[dir] = {};
                data[dir].pub = pubMsgsSec;
            }
        }

        if (fs.existsSync(subLogPath)) {
            const subMsgsSec = parseLogFile(subLogPath);
            if (subMsgsSec) {
                if (!data[dir]) data[dir] = {};
                data[dir].sub = subMsgsSec;
            }
        }
    }

    const sortedDirs = Object.keys(data).sort((a, b) => getSortKey(a) - getSortKey(b));

    console.log('| Sync Period | Pub r/s | Sub r/s |');
    console.log('|---|---|---|');

    for (const dirName of sortedDirs) {
        const pubRs = data[dirName].pub || 'N/A';
        const subRs = data[dirName].sub || 'N/A';
        console.log(`| ${dirName} | ${pubRs} | ${subRs} |`);
    }
}

main();

File: parser-py/parser.py

import re
import os

def parse_log_file(file_path):
    with open(file_path, 'r') as f:
        content = f.read()
        match = re.search(r'(\d{1,3}(?:,\d{3})*|\d+)\s+msgs/sec', content)
        if match:
            return match.group(1).replace(',', '')
    return None

def get_sort_key(dir_name):
    dir_name = dir_name.lower()
    if dir_name.endswith('ms'):
        return int(dir_name[:-2])
    elif dir_name.endswith('s'):
        return int(dir_name[:-1]) * 1000
    elif dir_name.endswith('m'):
        return int(dir_name[:-1]) * 60 * 1000
    else:
        return float('inf') # Should not happen with the given data

def main():
    data = {}
    for root, dirs, files in os.walk('.'):
        for file in files:
            if file.endswith('.log'):
                file_path = os.path.join(root, file)
                dir_name = os.path.basename(root)
                file_type = 'pub' if 'pub' in file else 'sub'
                msgs_sec = parse_log_file(file_path)
                if msgs_sec:
                    if dir_name not in data:
                        data[dir_name] = {}
                    data[dir_name][file_type] = msgs_sec

    # Sort data by directory name using the custom sort key
    sorted_dirs = sorted(data.keys(), key=get_sort_key)

    # Print Markdown table
    print('| Sync Period | Pub r/s | Sub r/s |')
    print('|---|---|---|')
    for dir_name in sorted_dirs:
        pub_rs = data[dir_name].get('pub', 'N/A')
        sub_rs = data[dir_name].get('sub', 'N/A')
        print(f"| {dir_name} | {pub_rs} | {sub_rs} |")

if __name__ == '__main__':
    main()
